jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -156,76 +154,65 @@ public boolean isSpatRecord() {
+				if (messageType == "MAP") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedMAPJson());
+				} else if (messageType == "TIM") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedTIMJson());
 				}
-
-				if (!(isSpatRecord() && msgMetadata instanceof OdeSpatMetadata
-        			&& !((OdeSpatMetadata) msgMetadata).getIsCertPresent())) {
-					if (checkHeader(msgPayload) == "Ieee1609Dot2Data") {
-						Asn1Encoding msgEncoding = new Asn1Encoding("root", "Ieee1609Dot2Data", EncodingRule.COER);
-						msgMetadata.addEncoding(msgEncoding);
-					}
-				}
-
-				Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-						EncodingRule.UPER);
-				msgMetadata.addEncoding(unsecuredDataEncoding);
-
-				publisher.publish(xmlUtils.toXml(odeData),
-						publisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
 			}
+
 			serialId.increment();
 		}
 	}
 
-	
 	/**
-		* Checks the header of the OdeMsgPayload and determines the encoding rule to be used in the Asn1DecoderInput XML.
-		* The payload is checked for various message start flags. It will add the encoding rule to the Asn1DecoderInput XML
-		* to tell the ASN1 codec to extract data from the header into the output message.
+		* Determines the message type based off the most likely start flag
 		* 
-		* @param payload The OdeMsgPayload to check the header of.
-		* @return The encoding rule to be used in the Asn1DecoderInput XML.
+		* @param payload The OdeMsgPayload to check the content of.
 		*/
-	public String checkHeader(OdeMsgPayload payload) {
-		JSONObject payloadJson;
-		String header = null;
+	public String determineMessageType(OdeMsgPayload payload) {
+		String messageType = "";
 		try {
-			payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
-			String hexPacket = payloadJson.getString("bytes");
-
-			for (String key : msgStartFlags.keySet()) {
-				String startFlag = msgStartFlags.get(key);
-				int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
-				logger.debug("Start index for " + key + "(" + startFlag + ")" + " is: " + startIndex);
-				if (startIndex <= 20 && startIndex != 0 && startIndex != -1) {
-					logger.debug("Message has supported Ieee1609Dot2Data header, adding encoding rule to Asn1DecoderInput XML");
-					header = "Ieee1609Dot2Data";
-					break;
+			JSONObject payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
+			String hexString = payloadJson.getString("bytes").toLowerCase();
+
+			HashMap<String, Integer> flagIndexes = new HashMap<String, Integer>();
+			flagIndexes.put("MAP", hexString.indexOf(msgStartFlags.get("MAP")));
+			flagIndexes.put("TIM", hexString.indexOf(msgStartFlags.get("TIM")));
+
+			int lowestIndex = Integer.MAX_VALUE;
+			for (String key : flagIndexes.keySet()) {
+				if (flagIndexes.get(key) == -1)
+					continue;
```

Comment:
This case is actually going to happen a lot. I suppose I can add a debug log but this will probably get printed 200+ times for every 100 line log file.

Reply:
Addressed

Reply:
Ah I thought it would happen less often; I'm fine with either including this debug log or not, then.

Reply:
Yeah the main reason is we just don't know what types of messages are in log files so we iterate through the possibilities for each one. If it isn't a match, this case triggers, so we continue onto the next iteration. Since it really is only between MAP and TIM at the moment, I suppose it could occur 0 times or once per message depending upon if it was a MAP or TIM.